src/EventLoop.php

```diff
@@ -0,0 +1,220 @@
+    }
+
+    /**
+     * Execute a callback when a stream resource becomes writable.
+     * 
+     * @param resource $stream The stream to monitor.
+     * @param callable $callback The callback to execute.
+     * 
+     * @return string An identifier that can be used to cancel, enable or disable the event.
+     */
+    public static function onWritable($stream, callable $callback)
+    {
+        return self::get()->onWritable($stream, $callback);
+    }
+
+    /**
+     * Execute a callback when a signal is received.
+     * 
+     * @param int $signo The signal number to monitor.
+     * @param callable $callback The callback to execute.
+     * 
+     * @return string An identifier that can be used to cancel, enable or disable the event.
+     */
+    public static function onSignal($signo, callable $callback)
+    {
+        return self::get()->onSignal($signo, $callback);
+    }
+
+    /**
+     * Execute a callback when an error occurs.
+     * 
+     * @param callable $callback The callback to execute.
+     * 
+     * @return string An identifier that can be used to cancel, enable or disable the event.
+     */
+    public static function onError(callable $callback)
+    {
+        return self::get()->onError($callback);
+    }
+
+    /**
+     * Enable an event.
+     * 
+     * @param string $eventIdentifier The event identifier.
+     * 
+     * @return void
+     */
+    public static function enable($eventIdentifier)
+    {
+        self::get()->enable($eventIdentifier);
+    }
+
+    /**
+     * Disable an event.
+     * 
+     * @param string $eventIdentifier The event identifier.
+     * 
+     * @return void
+     */
+    public static function disable($eventIdentifier)
+    {
+        self::get()->disable($eventIdentifier);
+    }
+
+    /**
+     * Cancel an event.
+     * 
+     * @param string $eventIdentifier The event identifier.
+     * 
+     * @return void
+     */
+    public static function cancel($eventIdentifier)
+    {
+        self::get()->cancel($eventIdentifier);
+    }
+
+    /**
+     * Reference an event.
+     * 
+     * This will keep the event loop alive whilst the event is still being monitored. Events have this state by default.
+     * 
+     * @param string $eventIdentifier The event identifier.
+     * 
+     * @return void
+     */
+    public static function reference($eventIdentifier)
+    {
+        self::get()->reference($eventIdentifier);
+    }
+
+    /**
+     * Unreference an event.
+     * 
+     * The event loop should exit the run method when only unreferenced events are still being monitored. Events are all
+     * referenced by default.
+     * 
+     * @param string $eventIdentifier The event identifier.
+     * 
+     * @return void
+     */
+    public static function unreference($eventIdentifier)
+    {
+        self::get()->unreference($eventIdentifier);
+    }
+
+    /**
+     * Disable construction as this is a static class.
+     */
+    public function __construct()
```

Comment:
The difference is more that IDEs can directly warn you instead of relying on runtime errors.